frameworks/keyed/incremental-dom/src/main.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * Incremental DOM is a virtual-dom diffing library; not a framework.
+ * See: http://google.github.io/incremental-dom/
+ */
+
+import { patch } from "incremental-dom";
+import { DataCollection } from "./data-collection";
+
+import { Button } from "./components/button";
+import { H1 } from "./components/header";
+import { Div } from "./components/div";
+import { GlyphIconSpan } from "./components/span";
+import { Table, TableRow } from "./components/table";
+
+
+let root: HTMLElement;
+let state: DataCollection = new DataCollection();
+let selectedId: number = null;
+
+function render() {
+  if (!root) root = document.getElementById('main')!;
+  patch(root, incrementalDom);
+}
+
+function actionRun() {
+  state.createRows(1000);
+  render();
+}
+
+function actionRunLots() {
+  state.createRows(10000);
+  render();
+}
+
+function actionAdd() {
+  state.appendRows(1000);
+  render();
+}
+
+function actionUpdate() {
+  state.mutate((e) => { e.label += ' !!! ' }, 10);
+  render();
+}
+
+function actionClear() {
+  state.clear();
+  render();
+}
+
+function actionSwapRows() {
+  state.swap(1, 998);
+  render();
+}
+
+function actionSelectRow(id: number) {
+  selectedId = id;
+  console.log(state.data.filter(d => d.id === id)[0]);

```

Comment:
This console log is probably polluting the benchmark

Reply:
Good catch! I was wondering why this benchmark in particular was doing so poorly. plot twist: It was me all along.